tools/depchecker/godepchecker.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// Dependency represents a Golang dependency
+type Import struct {
+	Name    string `yaml:"name"`
+	Version string `yaml:"version"`
+	Repo    string `yaml:"repo,omitempty"`
+}
+
+// Glide represents a glide yaml file
+type Glide struct {
+	Imports []Import `yaml:"imports"`
+}
+
+func readGlide(filename string) (map[string]Import, error) {
+	deps := make(map[string]Import)
+
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	glide := Glide{}
+	err = yaml.Unmarshal(f, &glide)
+	if err != nil {
+		return nil, err
+	}
+	for _, dep := range glide.Imports {
+		deps[dep.Name] = dep
+	}
+	return deps, nil
+}
+
+func main() {
+	s2i, err := readGlide("glide.lock")
+	if err != nil {
+		fmt.Printf("error: can't read glide.lock: %v\n", err)
+		os.Exit(1)
+	}
+
+	origin, err := readGlide("../origin/glide.lock")
+	if err != nil {
+		fmt.Printf("info: can't read ../origin/glide.lock: %v, not continuing\n", err)
+		return
```

Comment:
should either `os.Exit(1)` both here and above, or just return both here and above?

at first blush seems like it should be consistent

Reply:
updated